### ISSUE TITLE
Continued: typing.NamedTuple: no value for argument 'typename' (1.8.2)

### ISSUE THREAD

**Issue by TaysirTayyab** [PyCQA/pylint#1898]
Follow-up on #1418, `NamedTuple` class form still throwing errors in pylint 1.8.2 with astroid 1.6.

### Steps to reproduce
Use the below test file (copied directly from [PyCQA/astroid@1.6](https://github.com/PyCQA/astroid/blob/1.6/astroid/tests/unittest_brain.py#L759)):
```python
## test.py ##
from typing import NamedTuple

class Example(NamedTuple):
    mything: int

Example(mything=1)
```

### Current behavior
Pylint complains about the above.
```
$ pylint -E test.py
No config file found, using default configuration
************* Module test
E:  6, 0: No value for argument 'typename' in constructor call (no-value-for-parameter)
```

### Expected behavior
This is valid syntax and shouldn't throw an error.

### pylint --version output
```bash
$ pylint --version
No config file found, using default configuration
pylint 1.8.2,
astroid 1.6.1
Python 3.6.3 (default, Oct  4 2017, 06:09:15)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

**Comment by terrisgit** [comment]
Happens to me too

**Comment by PCManticore** [comment]
This is definitely weird, since I can't seem to reproduce it locally:
```
⋊> ~/p/p/playground python -m pylint a.py                                                    09:28:39
No config file found, using default configuration
************* Module a
C:  1, 0: Missing module docstring (missing-docstring)
C:  4, 0: Missing class docstring (missing-docstring)
R:  4, 0: Too few public methods (0/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 2.50/10 (previous run: 2.50/10, +0.00)

⋊> ~/p/p/playground pylint --version                                                         09:28:41
No config file found, using default configuration
pylint 1.8.2,
astroid 1.6.0
Python 3.6.3 (default, Jan  7 2018, 10:31:53)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
⋊> ~/p/p/playground
```
Does anyone have a docker container or something that I could use to reproduce this?

**Comment by eddie-dunn** [comment]
### Steps to reproduce

1. Create `reproduce.py` and `Dockerfile` in a directory of your choosing:

```
"""reproduce.py"""
from typing import NamedTuple

class Example(NamedTuple):
    mything: int

Example(mything=1)

```
```
# Dockerfile
FROM python:3.6

COPY reproduce.py reproduce.py
RUN pip install pylint

CMD pylint -E reproduce.py
```


2. Run the following:
```
docker build --build-arg https_proxy=$https_proxy -t pylint_test . && docker run pylint_test
```

(you may leave out `--build-arg https_proxy=$https_proxy` if you don't need it)

**Comment by TaysirTayyab** [comment]
@PCManticore: Verified @eddie-dunn's reproduce with specifically pylint 1.8.2 astroid 1.6.1.

## Reproduce
1. Make this file
```python
""" reproduce.py """
from typing import NamedTuple

class Example(NamedTuple):
    mything: int

Example(mything=1)
```
2. Run this, changing the path to `reproduce.py` to match your local machine.
```bash
docker run -it --rm -v "$PWD"/reproduce.py:/data/reproduce.py -w /data python:3.6 sh -c "pip3 install pylint==1.8.2 && pylint -E reproduce.py"
```

## Current Behavior
After running the docker command above, you'll see this (the pip install part is truncated)
```bash
#<<< some pip install stuff here >>>#
Installing collected packages: lazy-object-proxy, six, wrapt, astroid, isort, mccabe, pylint
Successfully installed astroid-1.6.1 isort-4.3.4 lazy-object-proxy-1.3.1 mccabe-0.6.1 pylint-1.8.2 six-1.11.0 wrapt-1.10.11
No config file found, using default configuration
************* Module reproduce
E:  6, 0: No value for argument 'typename' in constructor call (no-value-for-parameter)
```

**Comment by feluxe** [comment]
I'm affected too.

OS: Linux

```
$ pipenv run pylint --version 

No config file found, using default configuration
pylint 1.8.2, 
astroid 1.6.1
Python 3.6.4 (default, Jan  5 2018, 02:35:40) 
[GCC 7.2.1 20171224]
```

**Comment by orez-** [comment]
I spent today digging through the code to try to understand this issue better. It looks like the [`_explicit_inference`](https://github.com/PyCQA/astroid/blob/astroid-1.6.1/astroid/__init__.py#L112) of [`infer_typing_namedtuple_class`](https://github.com/PyCQA/astroid/blob/astroid-1.6.1/astroid/brain/brain_typing.py#L86-L89) being set on the `Example` node in @TaysirTayyab's example is being overwritten by [`_infer_object__new__decorator`](https://github.com/PyCQA/astroid/blob/astroid-1.6.1/astroid/brain/brain_builtin_inference.py#L516-L519) (notably, commenting out the registration of `_infer_object__new__decorator` seems to fix this example). I'm not sure in what order these transformations get loaded, but I suspect that might account for the inconsistent reproduction across envs.

It seems astroid 1.6.1 does not have the changes from PyCQA/astroid#489 , which applies these inferences more deliberately, and I think might fix this issue.

**Comment by brycepg** [comment]
@orez- Yeah looks like I forgot to backport that bug fix. Right now astroid master is for 2.0 (unreleased) and the `1.6` branch is now the a branch which only receives only bug fixes.

**Comment by TaysirTayyab** [comment]
@PCManticore updated astroid to 1.6.2 (to get PyCQA/astroid#513) and pylint no longer throwing errors for the reproduce. Believe this can be closed as fixed.

Thanks so much @brycepg @PCManticore!
